### ISSUE TITLE
Aspect Ratio Confusion

### ISSUE THREAD

**Issue by shoober420** [chocolate-doom/chocolate-doom#1181]
Version of Chocolate Doom: Latest git pull as of post

Operating System and version: Arch Linux 64bit

Game: Doom

I can’t help but notice how using aspect ratio correction makes the screen blurry. When disabled, it almost looks as sharp as crispy doom. Which led me to figuring out how to achieve the correct aspect ratio without enabling the setting. I came across this thread.

https://github.com/chocolate-doom/chocolate-doom/issues/968

It left me still puzzled. I use a CRT by the way. What I noticed was, when I set my full screen resolution to 640x400 and disable aspect ratio correction and scaling, the picture is very sharp, almost as sharp as crispy doom, but at native resolution. Spectres are harder to see using the native 320x200, as compared to crispy dooms 640x400.

I almost stuck with this, except now I don’t know if it’s rendering at the correct aspect ratio. I noticed the higher up I set my resolution with both aspect ratio correction and scaling disabled, the more blurry the screen got. This applies to if both settings are enabled as well.

So I’m asking advice from the man himself, fragglet. Honestly, anyone who can confirm my question is good at this point. I was trying to compare screenshots from my phone to my CRT, and it wasn’t making me feel as if it was legit and confirmed.

How can I achieve the correct aspect ratio while using a low resolution, preferably 640x400, with aspect ratio correction disabled to maintain the sharpness?

**Comment by chungy** [comment]
If you have a 4:3 CRT (almost all of them were), you should be able to use 640x400 with `aspect_ratio_correct 0`, and use the monitor's own controls to make sure the picture fills the screen. This is actually a perfectly good use case, just an unusual one for most people today (with 16:9 or 21:9 high-res LCD monitors). Doom originally output to the screen at 320x200, and by using 640x400 instead, you are just doubling the size of the pixels being output.

Depending on the higher resolutions you are setting, they may not be multiples of 320 and 200 and that's where blurring would become introduced as pixels need to be interpolated to fill in the space requested. You need to have whole multiples of both dimensions to get crisp outputs (320x200, 640x400, 960x600, 1280x800, etc).

**Comment by fragglet** [comment]
Thanks for asking - I recommend setting `integer_scaling` to `1` inside your `chocolate-doom.cfg` file.

Here's why: if you have a monitor with a vertical resolution that is not a multiple of 200 (eg. 1080p), "full screen" will give a resolution that is not an integer multiple of 320 or 200 in either the horizontal or vertical axes. That means that you will get some blurriness in each axis.

`integer_scaling=1` will force at least one axis to be an integer multiple of 320x200. For example in the case of 1080p you might end up with 1333x1000, only non-integer (blurry) in one axis. This can make a lot of difference if the blurriness is something bothers you - the only downside is that you get some windowboxing and it won't run completely full screen.

**Comment by shoober420** [comment]
Thanks guys. I’m one of those weird dudes who loves CRTs. Those new 240Hz LCDs look nice, but I will always keep a CRT around for use cases like retro games. I keep a CRT TV as well for this very purpose. My beloved Sony Trinitron lol. I think my IBM P275 has Trinitron tunes as well.

This cleared up a lot of questions. I didn’t know if using a 8:5 resolution (320x200 multiples) would brake the original aspect ratio intended. I also tried 1600x1200 as mentioned in that thread, but I found it a tad bit blurry, or blocky. It just didn’t look right compared to using 640x400. Your explanations clarified everything for me. Thank you sirs.

Good day fragglet and chunky.
